content/js/record.md

```diff
@@ -1,307 +1,331 @@
 
+const Note = skygear.Record.extend('note');
 const note = new Note({ 'content': 'Hello World' });
+
 ```
 
+**Important:** The skygear server will create a database schema for each record
+type based on the first record saved and if later records contains fields that doesn't
+exist in the current schema, a schema migration is performed.
+
+
 ### Record Database
 
-You will be provided with a private and a public database.
+You are provided with a public and private database:
+
+- Records saved in the public database are readable by everyone by default
+(including anonymous users). You may refine this by changing the
+[default access control rules](/js/guide/access-control#acl-default)
+or setting rules for each record independently.
+
+- Each user has their own private database where records are only accessible
+to themselves regardless of what access control rules are set on the records.
 
-- Everything in the private database is truly private, regardless of what access
-control entity you set to the record. In other words, each user has his own
-private database, and only himself has access to it.
-- Record saved at public database is by default public. Even without
-logging in, records in the public database can be queried (but not updated).
-To control the access, you may set different access control to the record.
-- The database objects can be accessed with `skygear.publicDB` and
+- The database interface can be accessed via `skygear.publicDB` and
 `skygear.privateDB`.
 
 <a name="basic-crud"></a>
 ## Basic CRUD
 
-### Creating a record
+### Creating records
 
-You can save a public record to server as the following.
+Saving a record to the public database:
 
 ``` javascript
-skygear.publicDB.save(new Note({
-  'content': 'Hello World!'
-})).then((record) => {
-  console.log(record);
-}, (error) => {
-  console.error(error);
-});
+const Note = skygear.Record.extend('note');
+const note = new Note({ 'content': 'Hello World!' });
+
+skygear.publicDB.save(note)
+  .then((saved_record) => {
+    console.log(saved_record);
+  })
+  .catch((error) => {
+    console.error(error);
+  });
 ```
 
-#### Batch save
-
-You can also batch save multiple records at one time.
+Saving multiple records to the public database:
 
 ``` javascript
-skygear.publicDb.save([goodNote1, goodNote2, badNote3, goodNote4, badNote5])
-.then((result) => {
-  console.log(result.savedRecords);
-  // [goodNote1, goodNote2, undefined, goodNote4, undefined]
-  console.log(result.errors);
-  // [undefined, undefined, error3, undefined, error5]
-}, (error) => { /* request error */ })
+const note1 = new Note({ date: '2016-09-07', content: 'first note' }),
+const note2 = new Note({ date: '2016-09-08', content: 'second note'}),
+const note3 = new Note({ date: new Date(),   content: 'third note' }),
+const note4 = new Note({ date: '2016-09-10', content: 'forth note' }),
+const note5 = new Note({ date: new Date(),   content: 'fifth note' }),
```

Comment:
Need more explanations on what this is illustrating and why some fail.
Perhaps first illustrate a simple successful example, followed by this complex example?